src/metabase/automagic_dashboards/core.clj

```diff
@@ -1047,21 +1057,30 @@
                                           :table_id   table-id}))))
        (get-in question [:dataset_query :query :aggregation])))
 
-(defn- collect-breakout-fields
+(s/defn ^:private collect-breakout-fields :- (s/maybe [(type Field)])
```

Comment:
The calling code definitely expects this to return a sequence of Fields. Because of bugs this would sometimes return `[nil]`. So the schema check here catches those bugs sooner